### PR TITLE
[1.7] Validate Gateway specs attempting to bind with <1024 port without root (#26699)

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/26708

### CODE REVIEW

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 1]
```diff
@@ -152,8 +151,72 @@ func deprecatedSettingsMessage(iop *v1alpha1.IstioOperatorSpec) string {
...
+			}
+			if p.TargetPort != nil && p.TargetPort.Type == intstr.Int {
+				tp = int(p.TargetPort.IntVal)
+			}
+			if tp == 0 && p.Port > 1024 {
+				// Target port defaults to port. If its >1024, it is safe.
+				continue
+			}
+			if tp < 1024 {
+				errs = util.AppendErr(errs, fmt.Errorf(format, p.Name, p.Port, gw.Name, tp, name))
```
Comment (mandarjog) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:215]: message: Here and in the other place, even if you have not set targetPort the message will read as if you had.

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 2]
```diff
@@ -152,8 +151,72 @@ func deprecatedSettingsMessage(iop *v1alpha1.IstioOperatorSpec) string {
...
+	return CheckServicePorts(values, spec)
+}
+
+// CheckServicePorts validates Service ports. Specifically, this currently
+// asserts that all ports will bind to a port number greater than 1024 when not
+// running as root.
+func CheckServicePorts(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
+	var errs util.Errors
+	if !values.GetGateways().GetIstioIngressgateway().GetRunAsRoot().GetValue() {
+		errs = util.AppendErrs(errs, validateGateways(spec.GetComponents().GetIngressGateways(), "istio-ingressgateway"))
```
Comment (mandarjog) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:164]: Are gateways defined by the user immune from this problem?

**Review by mandarjog — APPROVED** [review]
lgtm for telemetry.

I have two other comments, but they are non-blocking.

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**